client/internal/routemanager/manager.go

```diff
@@ -0,0 +1,183 @@
+package routemanager
+
+import (
+	"context"
+	"fmt"
+	"github.com/netbirdio/netbird/client/status"
+	"github.com/netbirdio/netbird/client/system"
+	"github.com/netbirdio/netbird/iface"
+	"github.com/netbirdio/netbird/route"
+	log "github.com/sirupsen/logrus"
+	"runtime"
+	"sync"
+)
+
+// Manager is an instance of a route manager
+type Manager struct {
+	ctx            context.Context
+	stop           context.CancelFunc
+	mux            sync.Mutex
+	clientNetworks map[string]*clientNetwork
+	serverRoutes   map[string]*route.Route
+	serverRouter   *serverRouter
+	statusRecorder *status.Status
+	wgInterface    *iface.WGIface
+	pubKey         string
+}
+
+// NewManager returns a new route manager
+func NewManager(ctx context.Context, pubKey string, wgInterface *iface.WGIface, statusRecorder *status.Status) *Manager {
+	mCTX, cancel := context.WithCancel(ctx)
+	return &Manager{
+		ctx:            mCTX,
+		stop:           cancel,
+		clientNetworks: make(map[string]*clientNetwork),
+		serverRoutes:   make(map[string]*route.Route),
+		serverRouter: &serverRouter{
+			routes:                   make(map[string]*route.Route),
+			netForwardHistoryEnabled: isNetForwardHistoryEnabled(),
+			firewall:                 NewFirewall(ctx),
+		},
+		statusRecorder: statusRecorder,
+		wgInterface:    wgInterface,
+		pubKey:         pubKey,
+	}
+}
+
+// Stop stops the manager watchers and clean firewall rules
+func (m *Manager) Stop() {
+	m.stop()
+	m.serverRouter.firewall.CleanRoutingRules()
+}
+
+func sendUpdateToClientNetwork(updateChannel chan routesUpdate, updateSerial uint64, routes []*route.Route) {
+	updateChannel <- routesUpdate{
+		updateSerial: updateSerial,
+		routes:       routes,
+	}
+}
+
+func (m *Manager) updateClientNetworks(updateSerial uint64, networks map[string][]*route.Route) {
+	select {
+	case <-m.ctx.Done():
+		log.Infof("not updating client network because context is done: %v", m.ctx.Err())
+		return
+	default:
+		for id, client := range m.clientNetworks {
+			_, found := networks[id]
+			if !found {
+				log.Debugf("stopping client network watcher, %s", id)
+				go client.stop()
+				delete(m.clientNetworks, id)
+			}
+		}
+
+		for id, routes := range networks {
+			watcher, found := m.clientNetworks[id]
+			if !found {
+				watcher = newClientNetworkWatcher(m.ctx, m.wgInterface, m.statusRecorder, routes[0].Network)
```

Comment:
should we call it clientNetwork instead of watcher?

Reply:
changed to clientNetworkWatcher to avoid type override